packages/kysely/src/mysql.ts

```diff
@@ -0,0 +1,148 @@
+import { Client, ClientConfig } from 'https://deno.land/x/mysql@v2.10.2/mod.ts';
+
+import {
+  Kysely,
+  DatabaseConnection,
+  QueryResult,
+  Driver,
+  Dialect,
+  CompiledQuery,
+  MysqlAdapter,
+  MysqlIntrospector,
+  MysqlQueryCompiler,
+} from 'https://cdn.jsdelivr.net/npm/kysely/dist/esm/index-nodeless.js';
+
+import { TransactionSettings } from 'https://cdn.jsdelivr.net/npm/kysely/dist/esm/driver/driver.js';
+import { extendStackTrace } from 'https://cdn.jsdelivr.net/npm/kysely/dist/esm/util/stack-trace-utils.js';
+import {
+  isObject,
+} from 'https://cdn.jsdelivr.net/npm/kysely/dist/esm/util/object-utils.js';
+
+
+export class MysqlDriver implements Driver {
+  readonly #connections = new WeakMap<Client, DatabaseConnection>();
+  #pool: Client;
+
+  constructor(client: Client) {
+    this.#pool = client;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  async init(): Promise<void> {}
+
+  async acquireConnection() {
+    let connection = this.#connections.get(this.#pool);
+
+    if (!connection) {
+      connection = new MysqlConnection(this.#pool);
+      this.#connections.set(this.#pool, connection);
+    }
+
+    return connection;
+  }
+
+  async beginTransaction(
+    connection: DatabaseConnection,
+    settings: TransactionSettings
+  ): Promise<void> {
+    if (settings.isolationLevel) {
+      // On MySQL this sets the isolation level of the next transaction.
+      await connection.executeQuery(
+        CompiledQuery.raw(
+          `set transaction isolation level ${settings.isolationLevel}`
+        )
+      );
+    }
+
+    await connection.executeQuery(CompiledQuery.raw('begin'));
+  }
+
+  async commitTransaction(connection: DatabaseConnection): Promise<void> {
+    await connection.executeQuery(CompiledQuery.raw('commit'));
+  }
+
+  async rollbackTransaction(connection: DatabaseConnection): Promise<void> {
+    await connection.executeQuery(CompiledQuery.raw('rollback'));
+  }
```

Comment:
Why is there transaction support? Transactions do not work in the context of sessions, because the underlying problem is that to access the same cell in the database concurrently. So if there is a race condition, you'd have to roll back the transaction—but we cannot retry it, because that would imply to run the downstream middleware again, which would mean you'd send all messages again.